test/state/state.cpp

```diff
@@ -117,19 +117,30 @@ evmc_message build_message(const Transaction& tx, int64_t execution_gas_limit) n
 }
 }  // namespace
 
+void finalize(
+    State& state, evmc_revision rev, const address& coinbase, std::optional<uint64_t> block_reward)
+{
+    if (block_reward.has_value())
+        state.touch(coinbase).balance += *block_reward;
+
+    if (rev >= EVMC_SPURIOUS_DRAGON)
+    {
+        std::erase_if(
+            state.get_accounts(), [](const std::pair<const address, Account>& p) noexcept {
+                const auto& acc = p.second;
+                return acc.erasable && acc.is_empty();
+            });
+    }
+}
+
 std::variant<TransactionReceipt, std::error_code> transition(
     State& state, const BlockInfo& block, const Transaction& tx, evmc_revision rev, evmc::VM& vm)
 {
     auto& sender_acc = state.get(tx.sender);
     const auto validation_result = validate_transaction(sender_acc, block, tx, rev);
 
     if (holds_alternative<std::error_code>(validation_result))
-    {
-        // Pre EIP-158 coinbase has to be touched also for invalid tx.
```

Comment:
Why is this not needed now?

Reply:
This emulates applying block reward of 0. Previously, I wasn't exactly sure why this is needed. Now the block reward is "properly" applied in the end.